### ISSUE TITLE
Reliability issue when using SDMMC & USB Serial together with STM32H750VBT6

### ISSUE THREAD

**Issue by pacman1945** [stm32duino/Arduino_Core_STM32#1390]
**Describe the bug**
With STM32H750VBT6 running STM32duino_STM32SD Cardinfo Example, there is a reliability problem if the serial output is set to USB Serial.  That is, by enabling USB Support "CDC (generic Serial supersede U(S)ART)".  If USB Serial is enabled, there is over 50% chance card.init() fails.

If I disabled USB Serial and use the Generic Serial (PA10 & PA9), the CardInfo Examples worked perfectly every time.  The reliability problem only occurs when running the Cardinfo with USB Serial.

**To Reproduce**
Install the latest version of STM32duino_STM32SD (v1.2.3) and FatFs (v2.0.3) via Arduino Library Manager.
Choose example STM32duino STM32SD->Cardinfo
Modify the baud rate to 115200
Enable: USB Support "CDC (generic Serial supersede U(S)ART)"
Run the Cardinfo example
Open the "Serial Monitor"

**Screenshots**
When it fails, the following message will be displayed.
```
Initializing SD card...initialization failed. Is a card inserted?
```

**Desktop (please complete the following information):**
 - OS: MacOS
 - Arduino IDE version: 1.8.8
 - STM32 core version: 2.0.0
 - Tools menu settings if not the default: Board part number: Generic H750VBTx
                                                                     USB Support "CDC (generic Serial supersede U(S)ART)
                                                                     Default for other items
 - Upload method: SWD

**Board (please complete the following information):**
STM32H750VBT6 from the below URL
https://www.aliexpress.com/item/1005001683272407.html?spm=a2g0o.productlist.0.0.637d1d300fFQ5d&algo_pvid=b237e656-d04b-422e-84af-8be8ed4042d9&algo_expid=b237e656-d04b-422e-84af-8be8ed4042d9-0&btsid=0bb0624116200479098028721ef47f&ws_ab_test=searchweb0_0,searchweb201602_,searchweb201603

**Comment by fpistm** [comment]
Yes I guess it is possible.

**Comment by pacman1945** [comment]
The default system clock config for H743VIT6 & H750VBT6 is using HSI.  That is, the clock is running only at 64MHz.  See the variant_generic.c under "H742V(G-I)(H-T)_H743V(G-I)(H-T)_H750VBT_H753VI(H-T)" for details.  I've already tested using HSE clock running at 400MHz, but the reliability issue still existed.

**Comment by fpistm** [comment]
So I'm not so mad 😉 
One other possibility is an issue linked to the board. Did you try to retry when the init failed at sketch level ?

**Comment by pacman1945** [comment]
The Cardinfo Example sketch is already doing retry indefinitely.  BY THE WAY, when it fails, the card.init() call would block for around 1 minute before returning.  Subsequent retry would always fail, but returned immediately without blocking.

```
  while(!card.init(SD_DETECT_PIN)) {
    if (!disp) {
      Serial.println("initialization failed. Is a card inserted?");
      disp = true;
    }
    delay(10);
  }
```

**Comment by fpistm** [comment]
Hi @pacman1945 
Please, could you test with #1393?

**Comment by pacman1945** [comment]
I've re-tested the Cardinfo Example with your new patch.  With the default generic_clock.c, the Cardinfo Example failed every single time.  First, I thought maybe 'cos the sysclock was at 480MHz.  Then, I modified it to 400MHz and it still didn't work, not even once.

Afterwards, I copied the generic_clock.c from previous version (that is, from the official 2.0.0 release) and it worked intermittently, just like before.  I've also tried change the sysclock to 400MHz, and it also worked intermittently, exactly as before.

Conclusion:
- With the patch from #1393 and the old generic_clock.c, the reliability issue remains the same.  That is, the Cardinfo example sometimes works, sometimes doesn't.
- With the patch from #1393 and the new generic_clock.c that came with it, the Cardinfo example didn't work at all, even with sysclock running at 400MHz.

**Comment by pacman1945** [comment]
Okay, I can now get the patch from #1393 and the new generic_clock.c that came with it to work intermittently by modifying the clock config as follows:
```
50c50
<   RCC_OscInitStruct.PLL.PLLN = 60;
---
>   RCC_OscInitStruct.PLL.PLLN = 50;
52c52
<   RCC_OscInitStruct.PLL.PLLQ = 5;
---
>   RCC_OscInitStruct.PLL.PLLQ = 4;
```
Previously, I only modified the sysclock to 400MHz by changing RCC_OscInitStruct.PLL.PLLN to 50.  This will result in SDMMC clock running at 160MHz.  This time, I also modified RCC_OscInitStruct.PLL.PLLQ to 4, so as to set the SDMMC clock to 200MHz.  When sysclock running at 400MHz and SDMMC clock running at 200MHz, I can use the new generic_clock.c and everything behave exactly like before.  That is, it works intermittently.  However, this new patch still didn't fix the issue.  

BY THE WAY, are you able to get a STM32750VBT6 for testing?

**Comment by fpistm** [comment]
> BY THE WAY, are you able to get a STM32H750VBT6 for testing?

Unfortunately, I don't have one.

As you mentioned

> Additional Info: I've performed the same test on a STM32H743VIT6 board and it worked perfectly.

My guess is it is linke to the HW, probably one issue listed in the errata sheet.

**Comment by fpistm** [comment]
Hi @pacman1945 
I've got a WeAct MiniH750VBT6 which has been added to the core.
So, I do the test with it and I have no issue with SDMMC and USB Serial. I've tested several time with the WeAct variant and the generic one. 
I've tested on the top of the main branch, I know some issue around H7 have been done so maybe one of them helps on this is your board which have an issue.

I close this issue as I could not reproduce.
